### PR TITLE
Hunt down invalid doc blocks

### PR DESCRIPTION
**Pull request by greg0ire**
It is OK to ignore some of the errors we get, but not this one.

### CODE REVIEW

**Comment by greg0ire** [comment]
:thinking: there are some more occurrences of `InvalidDocBlock` in the baseline. I'm going to try to remove them.

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 1]
```diff
@@ -1793,25 +1793,7 @@ protected function validateAndCompleteFieldMapping(array $mapping): array
...
-     *                   isCascadePersist: bool,
-     *                   isCascadeRefresh: bool,
-     *                   isCascadeMerge: bool,
-     *                   isCascadeDetach: bool,
-     *                   type: int,
-     *                   originalField: string,
-     *                   originalClass: class-string,
-     *                   ?orphanRemoval: bool
-     *               }
+     * @psalm-return AssociationMapping
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:1796]: Slightly out of scope, but doing this change allows me not to pollute the phpstan baseline too much.

**Comment by greg0ire** [comment]
ping @derrabus :slightly_smiling_face:

File: lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php [hunk 2]
```diff
@@ -922,6 +922,8 @@ private function cacheToArray(array $cacheMapping): array
             $usage = (int) constant('Doctrine\ORM\Mapping\ClassMetadata::CACHE_USAGE_' . $usage);
         }
 
+        assert($usage !== '0' && $usage !== '');
```
Comment (derrabus) [inline lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: Can we be sure? I think we should either map all falsy values to `null` or make sure falsy values other than null also trigger an `InvalidArgumentException`.

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php (hunk 1 above)
Comment (derrabus) [reply lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:1796]: Looks reasonable to me.

File: lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php (hunk 2 above)
Comment (greg0ire) [reply lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: I think you're right, but let's maybe not do this in a patch release?

Comment (derrabus) [reply lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: Is a suddenly failing `assert()` better in a patch release? 😬

Comment (greg0ire) [reply lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: Provided assertions are not enabled in production, I'd say yes.

Comment (SenseException) [reply lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: Not a doc block (as the title suggests) though, but in comparison it screams at dev runtime when something's wrong. :-)

Comment (greg0ire) [reply lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php]: OK, I'll remove that line and ignore whatever SA errors come up then.

**Review by derrabus — APPROVED** [review]

**Pull request merged**